### DISCUSSION TITLE
Generate file in folder, not a single file.

### DISCUSSION THREAD

**Discussion by pqv2210** [infinitered/ignite#2319] [Q&A]
Hi developer team,
When i run command `ignite generate component <component_name>` in v8(Maverick version), i received a `component_name.tsx` file in `../app/components/component_name.tsx`.
Unlike before, when I run this command, I get the folder containing that component (same as v6).
Ex: run `ignite generate component switch`, i will receive Switch.tsx in ../app/components/switch/switch.tsx
How can i dynamically create a component that has a folder that wraps it?
Thank you.

**Comment by ryskin** [comment]
I did the same mistake before and create an issue too ) you need to update CLI, it creates a component in a new folder, I just checked (see screenshot)

`npx ignite-cli generate component <component_name>`

<img width="416" alt="image" src="https://user-images.githubusercontent.com/3756722/204938474-f389051d-d8b9-4051-af82-e16467e6f5a9.png">

**Comment by pqv2210** [comment]
@ryskin i updated to ignite-cli v8.4.2 and run again, but still generate 1 file :D
![Screen Shot 2022-12-01 at 15 28 57 1](https://user-images.githubusercontent.com/32980987/205004215-b061e926-b04b-45a8-a491-4998f056d708.png)

**Reply by ryskin** [reply]
I think with npx and without nxp can run different versions try `ignite-cli generate component test` since it show new version

**Reply by ryskin** [reply]
@pqv2210 Did you try? I had the same issue 2 versions of CLI were installed on my mac.

**Reply by pqv2210** [reply]
@ryskin i tried, but not work. I think dev team changed code generate file, they removed generate folder part.

**Reply by frankcalise** [reply]
> @ryskin i tried, but not work. I think dev team changed code generate file, they removed generate folder part.

Yes, we removed the folder generation in v8 because this best represents the way we build projects at Infinite Red. We are currently investigating adding this enhancement (see #2285 and #2317). Of course, you could also extend the templates yourself in the `ignite` directory

**Reply by pqv2210** [reply]
@ryskin @frankcalise thank for your support.
